### PR TITLE
Added mergesort.java file in the Sorting folder.

### PR DESCRIPTION
**Pull request by adi-ch1234**
I have added mergesort.java file which shows merge sorting algorithm in java. Its easy to understand by anyone because I have included comments to explain every step. Its helpful for both requiring theory and its practical application because I have included the definition, steps involved, time complexity and the procedure of the algorithm in comments at the top of the program. Anyone can understand and implement merge sort algorithm in java very easily using my program.

### CODE REVIEW

**Comment by SumitBikram** [comment]
@adi-ch1234! Thanks👍🏼for showing interest in contributing to this repo.

File: Sorting/mergesort.java
Comment (SumitBikram) [inline Sorting/mergesort.java:1]: Kindly properly format the code.

**Review by SumitBikram — CHANGES_REQUESTED** [review]
Please commit again after proper indentation and java format!
👌🏼Everything else is fine for me.

**Comment by adi-ch1234** [comment]
Hi , thank you for suggesting the changes. I have submitted the file again with proper indentation and java format.

**Pull request closed without merging**